plugins/hydra_ax_sweeper/tests/test_ax_sweeper_plugin.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import math
+import os
+from typing import Any, Callable, List, Optional
+
+import pytest
+import yaml
+from omegaconf import DictConfig
+
+from hydra.core.plugins import Plugins
+from hydra.plugins.sweeper import Sweeper
+
+# noinspection PyUnresolvedReferences
+from hydra.test_utils.test_utils import TSweepRunner  # noqa: F401
+from hydra.test_utils.test_utils import SweepTaskFunction
+from hydra_plugins.hydra_ax_sweeper import AxSweeper
+
+
+def test_discovery() -> None:
+    """
+    Tests that this plugin can be discovered via the plugins subsystem when looking for Sweeper
+    :return:
+    """
+    assert AxSweeper.__name__ in [x.__name__ for x in Plugins.discover(Sweeper)]
+
+
+class AxSweepTaskFunction(SweepTaskFunction):
```

Comment:
didn't we say that you can just make it more flexible instead of extending it?

Reply:
We did extend `SweepTaskFunction ` to take an additional function parameter. `AxSweepTaskFunction` does one more thing - read the  file generated by AxSweeper. As of now, reading the contents of a file is not required for the other components hence a new class.

Reply:
Ah, this is not what I meant.
don't fake the return value.
you can have an explicit reading of that file in the caller. look how verify_output_directory is called in other tests.

Reply:
Are you talking about the [`verify_output`](https://github.com/facebookresearch/hydra/blob/83136c3d91ce715bf6e63e781946bd28b9a9d37e/examples/advanced/hydra_app_example/tests/test_example_app.py#L18) method? I could not find a `verify_output_dir` method. 

In the current implementation, the file is deleted after the `SweepTaskFunction` is executed. If I were to read the file after the execution, I would have to take care of deleting it. Faking the output seems to be less complicated.

Reply:
I am talking about the implementation of verify_dir_outputs, which inspect the output directory of the job.
It's used inside the sweep context, so after the sweep but before the cleanup:
```python
        with sweep:
            assert sweep.returns is not None
            assert len(sweep.returns[0]) == 2
            for i in range(2):
                job_ret = sweep.returns[0][i]
                assert job_ret.overrides == expected_overrides[i]
                assert job_ret.cfg == expected_conf[i]
                verify_dir_outputs(job_ret, job_ret.overrides)
```

Reply:
Got it. It does sound better than the current approach.